lib/index.ts

```diff
@@ -54,12 +84,13 @@ export {
   buildDepTreeFromFiles,
   PkgTree,
   DepType,
+  LockfileType,
 };
 
 async function buildDepTree(
```

Comment:
We should be determining the lockfile type in the `buildDepTree` function, given just the target and lockfile and some options. We shouldn't expect the lockfile type to be derived before calling our lib given we can easily determine this.

Reply:
I have another point of view - every caller of our library will already know, if they want to pass `yarn.lock` or `package-lock.json` (because they already had to read file, since  `buildDepTree` requires `string`. Therefore, any effort to derive type of a lockfile would be unnecessary and can add some performance overhead.

That's the reason why I added this functionality to `buildDepTreeFromFiles`, because it can be called with any path.

Reply:
I did not mean a caller within the library but me calling this lib from Registry, maybe I misunderstand but I the way I will be calling this lib from registry is 
` const depTree = await lockFileParser.buildDepTree(rawManifetFile.text, rawLockFile); ` which is the call signature we all agreed really early on.

Reply:
Hmm, IIRC, initial signature was `buildDepTree(rawManifestFile, rawLockFile, options`, which was moved to `buildDepTree(rawManifestFile, rawLockFile, includeDev = false` and I now added `buildDepTree(rawManifestFile, rawLockFile, includeDev = false, lockfileType = 'npm')`*.

My point is, that even in registry, I assume you'd already know, which file you wanna parse (since we are looking specifically for `package-lock.json` or `yarn.lock`). Therefore, adding a functionality to determine a type of a lockfile is superfluous. Or am I missing something?

\* implemented via if statement